src/main/java/io/github/christophermanahan/carnitas/Router.java

```diff
@@ -28,53 +30,40 @@ Router post(String uri, Function<HTTPRequest, HTTPResponse> handler) {
     }
 
     public HTTPResponse handle(HTTPRequest request) {
-        if (routeAdded(request)) {
-            return handleRoute(request);
-        } else if (allowedUri(request)) {
-            return methodNotAllowed(request);
-        } else {
-            return notFound();
-        }
-    }
-
-    private Route route(HTTPRequest request) {
-        return new Route(request.method(), request.uri());
-    }
-
-    private boolean routeAdded(HTTPRequest request) {
         return map.keySet().stream()
-          .anyMatch(route -> route.equals(route(request)));
+          .filter(equals(request))
+          .findFirst()
+          .map(map::get)
+          .map(handler -> handler.apply(request))
+          .orElseGet(handler(request));
     }
 
-    private HTTPResponse handleRoute(HTTPRequest request) {
-        return map.get(
-          map.keySet().stream()
-            .filter(route -> route.equals(route(request)))
-            .findFirst()
-            .get()
-        ).apply(request);
+    private Predicate<Route> equals(HTTPRequest request) {
```

Comment:
I was a little confused by this method, then I realized it was private. I might consider renaming it though. `equals` has a reserved meaning in Java. What is a name that matches the intent?

Reply:
Renamed to `matcherFor`